config/dependency.go

```diff
@@ -290,15 +290,43 @@ func getTerragruntOutput(dependencyConfig Dependency, terragruntOptions *options
 	return &convertedOutput, isEmpty, errors.WithStackTrace(err)
 }
 
+// Clone terragrunt options and update context for dependency block so that the outputs can be read correctly
+func cloneTerragruntOptionsForDependencyOutput(terragruntOptions *options.TerragruntOptions, targetConfig string) (*options.TerragruntOptions, error) {
+	targetOptions := terragruntOptions.Clone(targetConfig)
+	targetOptions.TerraformCliArgs = []string{"output", "-json"}
+
+	// DownloadDir needs to be updated to be in the context of the new config, if using default
+	_, originalDefaultDownloadDir, err := options.DefaultWorkingAndDownloadDirs(terragruntOptions.TerragruntConfigPath)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	// Using default, so compute new download dir and update
```

Comment:
I'm missing some context here... Why do you need to check for the default here?

Reply:
If you set the download dir to a non-standard location using `--terragrunt-download-dir`, then we want to directly use the one set on the original config instead of overriding that back to the default. However, since we don't have the CLI context object here, we can't detect that the user passed that CLI in in this function. So we instead rely on the heuristic that if the original is using the default download dir, then we should use the default download dir in the context of the target config.